Site Editor Details text update to "Quick Edit"

benazeer-ben opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Fixes https://github.com/WordPress/gutenberg/issues/66612

## Why?
The term 'Details' is not matching with the context. So updating it to 'Quick Edit'.

## How?
Changing the text 'Details' to 'Quick Edit'.

## Testing Instructions

- Go to the WordPress admin > Gutenberg > Experiments and enable the 'Quick Edit in DataViews' experiment.
- Go to the Site editor > Pages.
- Switch to the Table view or Grid view.
- In the data views top bar, observe the label of the button to toggle this panel will now display as 'Quick Edit'.
- Open the panel.
- Observe there's no heading at the top of the panel to communicate what this panel is about.


## Screenshots
|Before|After|
|-|-|
|![Screenshot 2024-12-18 at 1 49 32 PM (2)](https://github.com/user-attachments/assets/9b6a1f84-f935-4f1d-a01f-8e094cef439b)|<img width="187" alt="Screenshot 2024-12-18 at 1 51 33 PM" src="https://github.com/user-attachments/assets/d2a3fe6f-f221-4e86-811c-da08626d3d26" />|

benazeer-ben committed:
site editor details text update

benazeer-ben committed:
Quick edit labelmin side panel

afercia commented on packages/edit-site/src/components/post-edit/index.js:
```diff
@@ -178,6 +178,7 @@ export function PostEdit( { postType, postId } ) {
 			} ) }
 			label={ __( 'Post Edit' ) }
 		>
+			<h2>{ __( 'Quick Edit' ) }</h2>
```
It's best to use a Heading component instead of a hardcoded H2:
```
			<Heading level={ 2 } size={ 16 }>
				{ __( 'Quick Edit' ) }
			</Heading>
```

afercia reviewed (changes requested):
@benazeer-ben thanks for working on this.

I left a note about the hardcoded H2.

Also, now that the panel starts with a new H2 heading, [the post title](https://github.com/WordPress/gutenberg/blob/130b6e248da39012d3018862ef7153ce3ff81bef/packages/editor/src/components/post-card-panel/index.js#L108-L113) in the panel should be changed to a h3. Otherwise, the headings hierarchy wouldn't be ideal with two H2. Screenshot to illustrate:

![Screenshot 2025-01-14 at 12 45 18](https://github.com/user-attachments/assets/8db655c7-79db-4960-921d-b1980c720722)

I guess that would need some minor adjustments to the margins.

Mamaduka commented:
This will also need design feedback since adding a static header diverges from the original designs.

benazeer-ben committed:
Merge branch 'trunk' of https://github.com/benazeer-ben/gutenberg into site-editor-details-text-update

benazeer-ben committed:
Feedback changes

benazeer-ben commented:
Hi @Mamaduka 

Thank you for your feedback!

I’ve updated the code based on your suggestions:

Moved the 'Quick Edit' heading to the post-card-panel and applied the required margin for proper spacing.

Please review the changes. I’m happy to incorporate any additional design suggestions.
![Screenshot (1)](https://github.com/user-attachments/assets/d3ac07d1-59ac-4223-b6ed-c92645bf6adb)

Mamaduka commented on packages/editor/src/components/post-card-panel/index.js:
```diff
@@ -99,6 +100,13 @@ export default function PostCardPanel( {
 
 	return (
 		<VStack spacing={ 1 } className="editor-post-card-panel">
+			<Heading
+				level={ 2 }
+				size={ 16 }
+				className="editor-post-card-panel__heading"
+			>
+				{ __( 'Quick Edit' ) }
+			</Heading>
```
This `PostCardPanel` is also used inside the editor. Any changes to it affect editors globally. We should be careful and avoid visual regressions.

## Screenshot
![CleanShot 2025-01-15 at 16 44 22](https://github.com/user-attachments/assets/089a65d2-ddb2-4d14-8600-01135fef33a5)

benazeer-ben replied:
Yes! @Mamaduka 
That's right. Updated in last commit.

benazeer-ben committed:
Feedback changes - Updates limiting to edit site

benazeer-ben committed:
lint update

afercia commented:
I just discovered the `label` prop here doesn't do anything:

https://github.com/WordPress/gutenberg/blob/aec60be54dfeb45191bbf3bb4b284a486303da23/packages/edit-site/src/components/post-edit/index.js#L181-L186

It should be `title`. At this point, I'd use this `title` prop with a value `Quick Edit`; it already renders a H2 heading. I'd consider to show the heading also in the 'no selection' state and adjust the CSS:

![Screenshot 2025-01-17 at 10 35 54](https://github.com/user-attachments/assets/78e2f860-874c-4b97-bdda-9c9172b355e8)

benazeer-ben committed:
Feedback changes

benazeer-ben commented:
Thanks, @afercia, for pointing this out.

I have updated the label to "Title" and applied the required CSS to both the selection and no-selection states.

Please take a look and let me know if there are any further suggestions.

afercia commented:
Checking after last commits, I'd think the two headings should be vertically aligned:

![Screenshot 2025-01-23 at 12 02 50](https://github.com/user-attachments/assets/59a3068e-7463-42b5-ab47-42571422dbc3)

Reducing the top padding of `.edit-site-post-edit` to 6 pixels should make it.

There are two failing e2e tests that need to be adjusted:

`test/e2e/specs/editor/various/sidebar.spec.js`
This should be investigated. I think that at some point it expects to find one H2 heading and instead now it finds tow of them. Making the locator selectors more specific should fix it.

`test/e2e/specs/site-editor/page-list.spec.js`
This one fails because it still tries to locate the button with the old 'Details` label. Updating it to `Quick edit` should fix it.
See `await page.getByRole( 'button', { name: 'Details' } ).click();`.

Mamaduka commented:
Let's test these components in all editors. We don't want to introduce accidental design regressions when fixing a bug.

benazeer-ben committed:
Reduced top padding

benazeer-ben committed:
e2e test fix
